lib/shopify_api/rest/redirect.ex

```diff
@@ -0,0 +1,112 @@
+defmodule ShopifyAPI.REST.Redirect do
+  @moduledoc """
+  Shopify REST API Redirect resources
+  """
+
+  alias ShopifyAPI.AuthToken
+  alias ShopifyAPI.REST
+
+  @doc """
+  Get a list of all the redirects.
+
+  ## Example
+
+    iex> ShopifyAPI.REST.Redirect.all(auth)
+    {:ok, [] = redirects}
+  """
+  def all(%AuthToken{} = auth, params \\ [], options \\ []),
+    do: REST.get(auth, "redirects.json", params, options)
+
+  @doc """
+  Return a single redirect.
+
+  ## Example
+
+    iex> ShopifyAPI.REST.Redirect.get(auth, integer)
+    {:ok, %{} = redirect}
+  """
+  def get(%AuthToken{} = auth, redirect_id, params \\ [], options \\ []),
+    do:
+      REST.get(
+        auth,
+        "redirects/#{redirect_id}.json",
+        params,
+        Keyword.merge([pagination: :none], options)
+      )
+
+  @doc """
+  Return a count of redirects.
+
+  ## Example
+
+    iex> ShopifyAPI.REST.Redirect.count(auth)
+    {:ok, integer = count}
+  """
+  def count(%AuthToken{} = auth, params \\ [], options \\ []),
+    do:
+      REST.get(auth, "redirects/count.json", params, Keyword.merge([pagination: :none], options))
+
+  @doc """
+  Update a redirect.
+
+  ## Example
+
+    iex> ShopifyAPI.REST.Redirect.update(auth, map)
+    {:ok, %{} = redirect}
+  """
+  def update(%AuthToken{} = auth, %{"redirect" => %{} = redirect}) do
+    atomized_map =
```

Comment:
This kind of conversion from strings to atoms can be dangerous, re: https://til.hashrocket.com/posts/gkwwfy9xvw-converting-strings-to-atoms-safely

We haven't done this in the past, I am wondering what it gets us?

From my knowledge we can just pass a string or atomized map in here and it will work.

`%{foo: "bar"}` and `%{"foo" => "bar"}` both produce the same payload to Shopify.

Reply:
when writing up the tests for this module, i saw that the downstream body passed along actually contained two separate "id" keys (one from the "id" originally passed and one from the converted :id key) in the JSON passed to the Bypass server. This should "just work" as both values are the same & if the parser overwrites the original key with the second value it doesn't really matter since they're the same. However, this seems like it could lead to undefined behavior or some JSON nazi deciding that this isn't valid JSON & so these requests will not work... some such far-fetched scenario (the kind of which I've had to go through before). 

I'm not aware of any HWIA analog in Elixir, so this pattern effectively applies that to the outer scope of the nested Map passed in the args and assigned to `redirect` here. 

You raise a good point about the atom limit; in my mind, attributes passed to this method should be defined in the compiled application & ensuring that there's no shenanigans is a part of the validation process for incoming requests out on the edge of the app... but that's a big assumption. I'm sure that model doesn't cover all use-cases.

Ultimately, I do believe this implementation does have merit; however, it is more overhead & more logic that could fail. As long as Shopify's JSON parsers continue to behave in a similar fashion as they do today, then it's probably fixing a problem we don't have.

Reply:
Fair points, also not sure we want to go down the road of deciding which key to use if both atom and string keys are passed.

As you can see it gets strange.
```
iex(2)> Jason.encode!(%{:id => "foo", "id" => "bar"})
"{\"id\":\"foo\",\"id\":\"bar\"}"
iex(3)> Jason.encode!(%{:id => "foo", "id" => "bar"}) |> Jason.decode!()
%{"id" => "foo"}
```

My understanding of JSON is the decoder takes the first key and uses that. Which is looks like Jason does. But the JSON spec doesn't appear to specify order when encoding. :can-of-worms-emoji:

Reply:
It looks like one conversion loop got missed: https://github.com/pixelunion/elixir-shopifyapi/pull/335/files#diff-0f01b717435a56fa6ea43e798ffb4844R96-R99

Reply:
@tres Looks fantastic, thanks for putting up with us here. Contributions to this repo are still new!

Reply:
thanks, @hez! I appreciate the time and thought spent on the PRs.